crates/ruff_python_formatter/src/string/mod.rs

```diff
@@ -161,117 +124,124 @@ impl Ranged for AnyStringPart<'_> {
+        self
+    }
+}
+
+impl Format<PyFormatContext<'_>> for FormatStringPart<'_> {
     fn fmt(&self, f: &mut PyFormatter) -> FormatResult<()> {
-        let parent_docstring_quote_style = f.context().docstring();
-        let locator = f.context().locator();
-        let result = match self.layout {
-            StringLayout::Default => {
-                if self.string.is_implicit_concatenated() {
-                    in_parentheses_only_group(&FormatStringContinuation::new(self.string)).fmt(f)
-                } else {
-                    StringPart::from_source(self.string.range(), &locator)
-                        .normalize(
-                            self.string.quoting(&locator),
-                            &locator,
-                            f.options().quote_style(),
-                            parent_docstring_quote_style,
-                        )
-                        .fmt(f)
-                }
-            }
-            StringLayout::DocString => {
-                let string_part = StringPart::from_source(self.string.range(), &locator);
-                let normalized = string_part.normalize(
-                    Quoting::CanChange,
-                    &locator,
-                    // Per PEP 8 and PEP 257, always prefer double quotes for docstrings
-                    QuoteStyle::Double,
-                    parent_docstring_quote_style,
-                );
-                docstring::format(&normalized, f)
-            }
-            StringLayout::ImplicitConcatenatedStringInBinaryLike => {
-                FormatStringContinuation::new(self.string).fmt(f)
-            }
-        };
-        // TODO(dhruvmanila): With PEP 701, comments can be inside f-strings.
-        // This is to mark all of those comments as formatted but we need to
-        // figure out how to handle them. Note that this needs to be done only
-        // after the f-string is formatted, so only for all the non-formatted
-        // comments.
-        if let AnyString::FString(fstring) = self.string {
-            let comments = f.context().comments();
-            fstring.value.elements().for_each(|value| {
-                comments.mark_verbatim_node_comments_formatted(value.into());
-            });
+        match self.string_part {
+            AnyStringPart::String(part) => part.format().with_options(self.context).fmt(f),
+            AnyStringPart::Bytes(part) => part.format().fmt(f),
+            AnyStringPart::FString(part) => part.format().with_options(self.context).fmt(f),
         }
-        result
     }
 }
 
-struct FormatStringContinuation<'a> {
+pub(crate) struct FormatStringContinuation<'a> {
     string: &'a AnyString<'a>,
+    context: StringContext,
 }
 
 impl<'a> FormatStringContinuation<'a> {
-    fn new(string: &'a AnyString<'a>) -> Self {
-        Self { string }
+    pub(crate) fn new(string: &'a AnyString<'a>) -> Self {
+        Self {
+            string,
+            context: StringContext::default(),
```

Comment:
There are calls to `FormatStringContinuation::new` which uses the default context. Should we explicitly pass in the default context in those cases? I'm fine with either.

Reply:
To me this mainly depends if the default context is guaranteed to do the right thing or if forgetting to call `with_context` may assume the wrong quotes and mess things up. In other words: Is calling `with_context` optional (`with_context` might be fine) or is it always required (constructor argument). I think we should always provide it because it can't know the right quotes, can it?

Also, implicit concatenated strings can never be docstrings, right? We should then change the `FormatStringContinuation` to only take `quoting` as an option to encode this information in the types.

Reply:
This is resolved because `FormatStringContinuation` determines the quoting by itself through `AnyString::quoting`.